File: js/day38.js
import readline from 'readline'

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter a year: ',(input) => {
  const year = Number(input);
  
  if (year % 4 == 0 && year % 100 != 0 || year % 400 == 0){
    console.log(`${year} is a leap year.`);
  } else {
    console.log(`${year} is not a leap year.`)
  }

  rl.close();
});

File: python/day38.py
import calendar

year_input = input("Enter a year: ")

try:
  year = int(year_input)


  if calendar.isleap(year):
    print(f"{year} is a leap year.")
  else: 
    print(f"{year} is not a leap year.")

except ValueError:
  print("Error: Please enter a valid number for the year")